packages/tc-ui/src/primitives/number/__tests__/e2e-record-number.cyp.js

```diff
@@ -16,28 +17,100 @@ describe('End-to-end - record Number type', () => {
 		});
 	});
 
-	it('can record an integer', () => {
-		cy.get('input[name=someInteger]').type(someInteger);
-		save();
+	describe('int', () => {
 
-		cy.get('#code').should('have.text', code);
-		cy.get('#someString').should('have.text', someString);
-		cy.get('#someInteger').should('have.text', String(someInteger));
-	});
+		it('can record an integer', () => {
+			cy.get('input[name=someInteger]').type(someInteger);
+			save();
 
-	it('can not accept non-integer value', () => {
-		cy.get('input[name=someInteger]').type(someInteger / 7);
-		save();
-
-		cy.url().should('contain', '/MainType/create');
-		cy.get('.o-message__content-main').should(
-			'contain',
-			'Oops. Could not create MainType record for e2e-demo',
-		);
-		cy.get('.o-message__content-additional').should(
-			'contain',
-			`Invalid value \`${someInteger /
-				7}\` for property \`someInteger\` on type \`MainType\`: Must be a finite integer`,
-		);
-	});
+			cy.get('#code').should('have.text', code);
+			cy.get('#someString').should('have.text', someString);
+			cy.get('#someInteger').should('have.text', String(someInteger));
+		});
+
+		it('rejects floats', () => {
+			cy.get('input[name=someInteger]').type('0.5');
```

Comment:
why not `type(0.5)` as we are testing for float

Reply:
I was guessing that cypress can only type strings. When the form is submitted they are sent as strings anyway